snapcraft/_store.py

```diff
@@ -100,6 +101,119 @@ def _requires_login():
         raise
 
 
+@contextmanager
+def _requires_command_success(command, not_found_err=None, failure_err=None):
```

Comment:
It should also be using `shutil.which` ;-)

Agreed on renaming the exception as well.